src/metadata.rs

```diff
@@ -1,81 +1,628 @@
+    pub fn to_query_string(&self) -> &str {
+        match self {
+            MetricId::CommonName(s) | MetricId::Id(s) | MetricId::Hxl(s) => s,
+        }
+    }
+
+    /// Generate a polars Expr that will do
+    /// an exact match on the MetricId
+    pub fn to_polars_expr(&self) -> Expr {
+        col(&self.to_col_name()).eq(self.to_query_string())
+    }
 
-impl SourceDataRelease{
-    pub fn get_metric_details(&self,metric_id: &str)->Option<&MetricMetadata>{
-        self.available_metrics.iter().find(
-            |m| m.source_metric_id == metric_id
-        ) 
+    /// Generate a polars Expr that will generate
+    /// a regex search for the content of the Id
+    pub fn to_fuzzy_polars_expr(&self) -> Expr {
+        col(&self.to_col_name())
+            .str()
+            .contains(lit(self.to_query_string()), false)
     }
 }
 
-// TODO we might want to just pass the MetricMetaData rather than
-// having to 
-impl From<MetricMetadata> for MetricRequest{
-    fn from(value: MetricMetadata) -> Self {
-        MetricRequest { 
-            column: value.parquet_column_name.clone(),
-            file: value.metric_parquet_file_url.clone().unwrap()
-         }
+impl From<MetricId> for Expr {
+    fn from(value: MetricId) -> Self {
+        value.to_polars_expr()
     }
 }
 
-impl From<&MetricMetadata> for MetricRequest{
-    fn from(value: &MetricMetadata) -> Self {
-        MetricRequest { 
-            column: value.parquet_column_name.clone(), 
-            file: value.metric_parquet_file_url.clone().unwrap()  
+impl Default for CountryMetadataPaths {
+    fn default() -> Self {
+        Self {
+            // TODO move this to enviroment variable or config or something
+            base_url: "https://popgetter.blob.core.windows.net/popgetter-dagster-test/test_2"
+                .into(),
+            geometry: "geometry_metadata.parquet".into(),
+            metrics: "metric_metadata.parquet".into(),
+            country: "country_metadata.parquet".into(),
+            source_data: "source_data_releases.parquet".into(),
+            data_publishers: "data_publishers.parquet".into(),
         }
     }
 }
 
-pub fn load_metadata(path: &str) -> Result<SourceDataRelease> {
-    let file = File::open(path)?;
-    let reader = BufReader::new(file);
-    let release: SourceDataRelease = serde_json::from_reader(reader)?;
-    Ok(release)
+/// `CountryMetadataLoader` takes a country iso string
+/// along with a CountryMetadataPaths and provides methods
+/// for fetching and construting a `Metadata` catalogue.
+pub struct CountryMetadataLoader {
+    country: String,
+    paths: CountryMetadataPaths,
+}
+
+/// A structure that represents a full joined lazy data frame
+/// containing all of the metadata
+pub struct ExpandedMetdataTable(pub LazyFrame);
+
+impl ExpandedMetdataTable {
+    /// Get access to the lazy data frame
+    pub fn as_df(&self) -> LazyFrame {
+        self.0.clone()
+    }
```

Comment:
I wonder if we can borrow the LazyFrame here - could the clone be expensive? It might be worth adding two distinct methods here aligned with the ad hoc conventions mentioned [here](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv). Alternatively, if we will always need to return an owned `LazyFrame` we could just have `.to_df()`:
```suggestion
    /// Gets reference to the lazy data frame
    pub fn as_df(&self) -> &LazyFrame {
        &self.0
    }
    /// Returns owned lazy data frame
    pub fn to_df(&self) -> LazyFrame {
        self.0.clone()
    }
```

Reply:
Ah ignore this suggestion, I missed your [comment](https://github.com/Urban-Analytics-Technology-Platform/popgetter-cli/pull/24/files#r1630868462) above!